app/Bundle.php

```diff
@@ -116,10 +116,19 @@ public function alterVouchers(Collection $vouchers, array $codes = [], Bundle $b
             $errors["codes"] = $missingCodes;
         }
 
-        // try to Run the vouchers we know are in the DB
+        // Try to run the vouchers we know are in the DB
         $vouchers->each(
-            function (Voucher $voucher) use ($bundle) {
-                $voucher->bundle()->associate($bundle)->save();
+            // Passing a pointer to $errors using "&", so we can change it, inside the loop.
+            // Otherwise the variable is immutable.
+            function (Voucher $voucher) use ($bundle, &$errors) {
+                // Check the voucher isn't disbursed, because we can't change those.
+                if ($voucher->bundle && $voucher->bundle->disbursed_at !== null) {
+                    // Throw it into an error.
+                    $errors["codes"][] = $voucher->code;
+                } else {
+                    // Change it's bundle
```

Comment:
```suggestion
                    // Change its bundle
```

Also the comment above may be better to say "Check if the voucher is disbursed" - the if is true for disbursed vouchers.